tests/json_schema.go

```diff
@@ -11,13 +12,266 @@ import (
 
 	"github.com/elastic/apm-server/processor"
 	"github.com/elastic/apm-server/tests/loader"
+	"github.com/elastic/beats/libbeat/common"
 )
 
+type JsonSchemaTestData struct {
+	Proc             processor.Processor
+	ValidPayloadPath string
+
+	// required keys in the json schema
+	RequiredKeys *Set
+	// coniditonally required keys in the json schema
+	ConditionalRequiredKeys map[string]Condition
+
+	// attributes defined as keywords in the ES template,
+	// but that do not require a length restriction in the json schema
+	KeywordExceptionKeys *Set
+	// some fields are nested or named differently on ES level
+	// than on intake API
+	TemplateToSchemaMapping map[string]string
+	LoadTemplatePaths       []string
+
+	TestData []TestData
+}
+
+type TestData struct {
+	Key       string
+	Invalid   []interface{}
```

Comment:
Don't we lose the checks for why the data is invalid with this?  Not that comparing strings was great but it was actually useful